packages/remix-react/browser.tsx

```diff
@@ -381,6 +411,80 @@ export function RemixBrowser(_props: RemixBrowserProps): ReactElement {
     });
   }, [location]);
 
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  let fogOfWarAbortControllerRef = React.useRef<AbortController | undefined>();
+
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  React.useEffect(() => {
+    // Don't perform any prefetching if the app asked not to or the user has saveData enabled
+    if (discover !== "render" || navigator.connection?.saveData === true) {
+      return;
+    }
+
+    async function fetchPatches() {
+      if (fogOfWarAbortControllerRef.current) {
+        fogOfWarAbortControllerRef.current.abort();
+      }
+
+      let lazyPaths = getFogOfWarPaths();
+      if (lazyPaths.length === 0) {
+        return;
+      }
```

Comment:
Short circuit there are no actively rendered paths to fetch - which means that they are all either matchable via the currently known routes, or they are a known 404 from a prior fetch.